connectors/utils.py

```diff
@@ -399,3 +401,60 @@ async def func_to_execute(*args, **kwargs):
         return func_to_execute
 
     return wrapper
+
+
+def ssl_context(certificate):
+    """Convert string to pem format and create a SSL context
+
+    Args:
+        certificate (str): certificate in string format
+
+    Returns:
+        ssl_context: SSL context with certificate
+    """
+    certificate = certificate.replace(" ", "\n")
+    certificate = " ".join(certificate.split("\n", 1))
+    certificate = " ".join(certificate.rsplit("\n", 1))
+    ctx = ssl.create_default_context()
+    ctx.load_verify_locations(cadata=certificate)
+    return ctx
+
+
+def encode(original_string):
+    """Performs encoding on the objects
+    containing special characters in their url, and
+    replaces single quote with two single quote since quote
+    is treated as an escape character
+
+    Args:
+        original_string(string): String containing special characters
+
+    Returns:
+        encoded_string(string): Parsed string without single quotes
+    """
+    encoded_string = urllib.parse.quote(original_string, safe="'")
+    return encoded_string.replace("'", "''")
+
+
+def get_expires_at(expires_in):
+    """Adds seconds to the current utc time.
+
+    Args:
+        expires_in (int): Seconds after which the token will expire.
+    """
+    expires_at = datetime.utcnow() + timedelta(seconds=expires_in)
+    # account for clock skew
+    expires_at -= timedelta(seconds=20)
+    return iso_utc(when=expires_at)
+
+
+def is_expired(expires_at):
+    """Compares the given time with present time
+
+    Args:
+        expires_at (datetime): Time of token expires.
+    """
+    # Recreate in case there's no expires_at present
+    if expires_at is None:
```

Comment:
This might lead to unexpected issues (I'm assuming all engineers can use this method because it's in utils) because when I pass `None` I would expect an exception rather than a positive outcome. 

It seems like the method signature is tight to the Sharepoint connector (see comment on line 457) so it's not a good idea to keep the connector's specific logic in utils.

Reply:
I got a comment to add it in utils.
 https://github.com/elastic/connectors-python/pull/537#discussion_r1121183957